src/adapter.cpp

```diff
@@ -1016,9 +1019,12 @@ SPid *find_pid(int aid, int p) {
     if (!ad)
         return NULL;
 
-    for (i = 0; i < MAX_PIDS; i++)
-        if ((ad->pids[i].flags > 0) && (ad->pids[i].pid == p))
+    for (i = 0; i < MAX_PIDS; i++) {
+        if ((ad->pids[i].flags > PID_STATE_INACTIVE) && (ad->pids[i].pid == p))
             return &ad->pids[i];
+        else if (ad->pids[i].flags == PID_STATE_INACTIVE)
```

Comment:
Is this due to the new sorting? Maybe add a comment on top of the loop that explains that since the PIDs are sorted we don't need to look past the first inactive one found.